report/topology.go

```diff
@@ -145,8 +145,10 @@ func (n Node) WithSets(sets Sets) Node {
 // WithMetric returns a fresh copy of n, with metric merged in at key.
 func (n Node) WithMetric(key string, metric Metric) Node {
 	result := n.Copy()
-	existing := n.Metrics[key]
-	result.Metrics[key] = existing.Merge(metric)
+	if existing, ok := n.Metrics[key]; ok {
+		metric = existing.Merge(metric)
+	}
+	n.Metrics[key] = metric
```

Comment:
(not strictly shadowed, but reused) but yeah, non obvious.

I'll add nil-value handling in for metrics, then we can remove this little if.